### YamlMime:UniversalReference
api_name: []
items:
- children:
  - azure.durable_functions.models.Task.action
  - azure.durable_functions.models.Task.exception
  - azure.durable_functions.models.Task.id
  - azure.durable_functions.models.Task.is_completed
  - azure.durable_functions.models.Task.is_faulted
  - azure.durable_functions.models.Task.result
  - azure.durable_functions.models.Task.timestamp
  class: azure.durable_functions.models.Task
  fullName: azure.durable_functions.models.Task
  inheritance:
  - type: builtins.object
  langs:
  - python
  module: azure.durable_functions.models
  name: Task
  summary: 'Represents some pending action.


    Similar to a native JavaScript promise in

    that it acts as a placeholder for outstanding asynchronous work, but has

    a synchronous implementation and is specific to Durable Functions.


    Tasks are only returned to an orchestration function when a

    [[DurableOrchestrationContext]] operation is not called with *yield*. They

    are useful for parallelization and timeout operations in conjunction with

    Task.all and Task.any.'
  syntax:
    content: Task(is_completed, is_faulted, action, result=None, timestamp=None, id_=None,
      exc=None, is_played=False)
  type: class
  uid: azure.durable_functions.models.Task
- class: azure.durable_functions.models.Task
  fullName: azure.durable_functions.models.Task.action
  langs:
  - python
  module: azure.durable_functions.models
  name: action
  summary: 'Get the scheduled action represented by the task.


    _Internal use only._'
  syntax: {}
  type: attribute
  uid: azure.durable_functions.models.Task.action
- class: azure.durable_functions.models.Task
  fullName: azure.durable_functions.models.Task.exception
  langs:
  - python
  module: azure.durable_functions.models
  name: exception
  summary: 'Get the error thrown when attempting to perform the task''s action.


    If the Task has not yet completed or has completed successfully, *None*'
  syntax: {}
  type: attribute
  uid: azure.durable_functions.models.Task.exception
- class: azure.durable_functions.models.Task
  fullName: azure.durable_functions.models.Task.id
  langs:
  - python
  module: azure.durable_functions.models
  name: id
  summary: 'Get the ID number of the task.


    _Internal use only._'
  syntax: {}
  type: attribute
  uid: azure.durable_functions.models.Task.id
- class: azure.durable_functions.models.Task
  fullName: azure.durable_functions.models.Task.is_completed
  langs:
  - python
  module: azure.durable_functions.models
  name: is_completed
  summary: 'Get indicator whether the task has completed.


    Note that completion is not equivalent to success.'
  syntax: {}
  type: attribute
  uid: azure.durable_functions.models.Task.is_completed
- class: azure.durable_functions.models.Task
  fullName: azure.durable_functions.models.Task.is_faulted
  langs:
  - python
  module: azure.durable_functions.models
  name: is_faulted
  summary: Get indicator whether the task faulted in some way due to error.
  syntax: {}
  type: attribute
  uid: azure.durable_functions.models.Task.is_faulted
- class: azure.durable_functions.models.Task
  fullName: azure.durable_functions.models.Task.result
  langs:
  - python
  module: azure.durable_functions.models
  name: result
  summary: Get the result of the task, if completed. Otherwise *None*.
  syntax: {}
  type: attribute
  uid: azure.durable_functions.models.Task.result
- class: azure.durable_functions.models.Task
  fullName: azure.durable_functions.models.Task.timestamp
  langs:
  - python
  module: azure.durable_functions.models
  name: timestamp
  summary: Get the timestamp of the task.
  syntax: {}
  type: attribute
  uid: azure.durable_functions.models.Task.timestamp
references:
- fullName: azure.durable_functions.models.Task.action
  isExternal: false
  name: action
  parent: azure.durable_functions.models.Task
  uid: azure.durable_functions.models.Task.action
- fullName: azure.durable_functions.models.Task.exception
  isExternal: false
  name: exception
  parent: azure.durable_functions.models.Task
  uid: azure.durable_functions.models.Task.exception
- fullName: azure.durable_functions.models.Task.id
  isExternal: false
  name: id
  parent: azure.durable_functions.models.Task
  uid: azure.durable_functions.models.Task.id
- fullName: azure.durable_functions.models.Task.is_completed
  isExternal: false
  name: is_completed
  parent: azure.durable_functions.models.Task
  uid: azure.durable_functions.models.Task.is_completed
- fullName: azure.durable_functions.models.Task.is_faulted
  isExternal: false
  name: is_faulted
  parent: azure.durable_functions.models.Task
  uid: azure.durable_functions.models.Task.is_faulted
- fullName: azure.durable_functions.models.Task.result
  isExternal: false
  name: result
  parent: azure.durable_functions.models.Task
  uid: azure.durable_functions.models.Task.result
- fullName: azure.durable_functions.models.Task.timestamp
  isExternal: false
  name: timestamp
  parent: azure.durable_functions.models.Task
  uid: azure.durable_functions.models.Task.timestamp
